lib/inputstreamhelper/__init__.py

```diff
@@ -136,7 +93,7 @@ def _diskspace(cls):
     @classmethod
     def _temp_path(cls):
         ''' Return temporary path, usually ~/.kodi/userdata/addon_data/script.module.inputstreamhelper/temp '''
-        temp_path = xbmc.translatePath(os.path.join(ADDON.getSetting('temp_path'), 'temp'))
+        temp_path = xbmc.translatePath(os.path.join(get_setting('temp_path', 'special://masterprofile/addon_data/script.module.inputstreamhelper'), 'temp'))
```

Comment:
I just noticed this bug while testing:
xbmc.translatePath doesn't support unicode in Python2, we need to add `to_unicode()` to support unicode paths in Python 2

```python
to_unicode(xbmc.translatePath(os.path.join(get_setting('temp_path', 'special://masterprofile/addon_data/script.module.inputstreamhelper'), 'temp')))
```

Reply:
Maybe good to add a `translate_path()` function that covers this correctly.

Reply:
I noticed most calls were already doing this.

Reply:
Fixed.